padl/dumptools/symfinder.py

```diff
@@ -736,6 +805,24 @@ def _find_branch(tree, lineno, source):
     return [tree]
 
 
+def find_scopedname_in_ipython(scoped_name: ScopedName) ->Tuple[ScopedName, ast.AST]:
+    """Find ScopedName in ipython
+
+    :param scoped_name:
+    :return:
+    """
+    source = node = None
+    for cell in sourceget._ipython_history()[::-1]:
+        try:
+            source, node, name = find_scopedname_in_source(scoped_name, cell)
+        except (NameNotFound, SyntaxError):
+            continue
+        break
+    if source is None:
+        raise NameNotFound(f'{",".join([str((var, n)) for var, n in scoped_name.variants])} not found.')
+    return source, node, name
+
+
```

Comment:
@jasonkhadka Do we need both `find_in_ipython` and `find_scopedname_in_ipython`?

Reply:
I am not sure yet, 
I don't see another use at the moment for `find_in_ipython`, but it is useful function so could be used or used for debugging.

Reply:
I guess in general it looks like there are a few functions that look almost duplicated like `find_scopedname_in_module` and `find_in_module`, and `find_scopedname_in_source` and `find_in_source`. Is it possible to just use the new `find_scopedname_` versions?

Reply:
Yes, I updated the finders to use `find_scopedname*` instead of `find_*`, cause it contains more information. I am not sure about throwing out `find_*`s yet. They can be used in debugging at the least and perhaps will be more useful for other new features. Also, it is not a whole lot of code.

Reply:
I refactored this.